scrape_repo_information.py

```diff
@@ -0,0 +1,107 @@
+
+Pair = collections.namedtuple("Pair", "lg1 lg2 last_updated created direction repo stems")
+types = ["trunk", "incubator", "nursery", "staging"]
+
+def get_bidix_stem_count(uri):
+    try:
+        dix_content = urllib.request.urlopen(uri).read().decode()
+        tree = xml.fromstring(dix_content)
+    except:
+        return -1 # FIXME: error handling
+
+    return len(tree.findall("*[@id='main']/e//l"))
+
+def main():
+    pairs = []
+    params = urllib.parse.urlencode(dict({"client_id": CLIENT_ID, "client_secret": CLIENT_SECRET}))
+    for repo_name in types:
+        types_data_url = "https://api.github.com/repos/apertium/apertium-%s/contents?" % repo_name
+        types_data_data = urllib.request.urlopen(types_data_url + params)
+        types_data = json.loads(types_data_data.read())
+        for type_content in types_data:
+            if "apertium" in type_content["name"] and (type_content["name"].count("-") == 2):
+                direction = ""
+                lang_pair_name = type_content["name"]
+                #getting names
+                _, lg1, lg2 = lang_pair_name.split("-")
+                if lg1 != lg2:
+                #getting into repository
+                    link = "https://api.github.com/repos/apertium/%s/contents?" % lang_pair_name
+                    repo_json = json.loads(urllib.request.urlopen(link + params).read())
+                    for el in repo_json:
+                        if el["name"] == "modes.xml":
+                            download_url = el["download_url"]
+                            html_for_blob_utf_8 = str((urllib.request.urlopen(download_url)).read(), "utf-8")
+                            tree = xml.fromstring(html_for_blob_utf_8)
+                            for mode in tree:
+                                if lg1+"-"+lg2 == mode.attrib["name"] and "<" not in direction:
+                                    direction += "<"
+                                if lg2+"-"+lg1 in mode.attrib["name"] and ">" not in direction:
+                                    direction += ">"
+
+                        elif el["name"] == "apertium-%s.%s.dix" % ((lg1+"-"+lg2), (lg1+"-"+lg2)):
+                            stems = get_bidix_stem_count(el["download_url"])
+
+                        if direction == "><":
+                            direction = "<>"
+
+                    first_page_commits_link = "https://api.github.com/repos/apertium/%s/commits?" % lang_pair_name
+                    first_page_commits_resp = urllib.request.urlopen(first_page_commits_link + params)
+                    first_page_commits_data = first_page_commits_resp.read()
+                    first_page_commit_json = json.loads(first_page_commits_data)
+
+                    if first_page_commits_resp.info().get("Link") is not None:
+                        number_of_pages = first_page_commits_resp.info().get("Link").split(",")[1].split("page=")[1].split(">")[0]
```

Comment:
I asked for a comment here that shows the format that is being parsed but the comment seems to be gone now.